src/BinSkim.Rules/RulesExtensionMethods.cs

```diff
@@ -35,7 +35,7 @@ public static string CreateOutputCoalescedByCompiler(this IList<ObjectModuleDeta
 
                 if (!libraries.TryGetValue(modulesKey, out SortedSet<string> modules))
                 {
-                    modules = libraries[modulesKey] = new SortedSet<string>();
+                    modules = libraries[modulesKey] = new SortedSet<string>(Comparer<string>.Create((a, b) => Path.GetFileNameWithoutExtension(a).CompareTo(Path.GetFileNameWithoutExtension(b))));
```

Comment:
Without this code change, under .net6.0 testing, the expected .sarif file and the actual .sarif file only differs in file **order**, as shown below:
![image](https://user-images.githubusercontent.com/119978670/208781184-bc6fc2c3-0a0b-4cd8-9b31-c976e4b55d3c.png)
It turns out that net6.0 changed the output of **String.Compare("dll_dllmain_stub.obj", "dll_dllmain.obj")**  compared with netcore3.1. The statement returns 1 in netcore3.1 while it returns -1 in net6.0. 

The code change only compares the filename **without extension**, so that the actual file can be consistent across freamworks.

Reply:
Can you please comment this code?

Reply:
Behavior changes when comparing strings in .NET 5 and later (official reprint) (iditect.com)
https://iditect.com/article/behavior-changes-when-comparing-strings-in-net-5-and-later-official-reprint.html

Add the following item in the project file to revert back using NLS.
![image](https://user-images.githubusercontent.com/119978670/209236044-f1f25f91-9377-495e-b099-a57111654fde.png)